### PR TITLE
Feature/fix test

### PR DESCRIPTION
**Pull request by DavoCoder**
- Addition of tests for all routes and possibilities: metrics, benefits, sdlc; using connected-apps, specific collectors, covering exception scenarios. Test coverage total 98.8 % - 2 routes pending (rolegroups / arm old calls) + APIManager pagination
- Non breaking refactor and adjustments of SDLC metrics (get metrics and not implemented loaders)
- Non breaking refactor of poller.xml
- Non breaking removal of non used flows and files

### CODE REVIEW

**Review by melamedjoel — APPROVED** [review]
I've fixed a minor typo. LGTM, tested, worked fine, and good to go

File: pom.xml [hunk 1]
```diff
@@ -71,6 +71,31 @@
...
 						</formats>
+						<ignoreFlows>
+							<!-- Ignoring the entry points, main listener, console, 
+							embedded dashboard listener -->     						
+      						<ignoreFlow>api-main</ignoreFlow>
+      						<ignoreFlow>api-console</ignoreFlow>
+      						<ignoreFlow>web-main</ignoreFlow>
+      						
+      						<!-- Ignoring the following 2 flows since they were never implemented properly, 
+      						they are not currently used but keeping them for future enhancements -->
```
Comment (panizzag) [inline pom.xml:82]: Nevermind, I just saw the file was deleted.

File: src/main/mule/sources/external/systems/apis/api-call-confluence.xml [hunk 2]
```diff
@@ -32,7 +32,7 @@ output application/json
 vars.outputPayload default [] ++ payload.results]]></ee:set-variable>
 					</ee:variables>
 				</ee:transform>
-				<set-variable value="#[output application/java --- if(payload.size == payload.limit) vars.counter else 0]" doc:name="Update Counter" doc:id="873c8284-6c10-4883-b898-00f920b3f235" variableName="counter"/>
+				<set-variable value="#[output application/java --- if(payload.size == payload.limit) 0 else vars.counter]" doc:name="Update Counter" doc:id="873c8284-6c10-4883-b898-00f920b3f235" variableName="counter"/>
```
Comment (panizzag) [inline src/main/mule/sources/external/systems/apis/api-call-confluence.xml:35]: This pagination logic deserves a deeper review. This change is fine but the pagination is half baked. For instance, https://developer.atlassian.com/server/confluence/pagination-in-the-rest-api/   shows how we can modify the paging setting a limit and even a start (offset). In the current configuration, this is embedded in the URL path. It should be separated as follows:

```
# SDLC Metrics
sdlc:
 ...
  confluence:
    enabled: "false" #Set this to true if confluence metrics is required
    host: "test.atlassian.net"
    port: "443"
    path: "/wiki/rest/api/content/search?next=true&cql=%28type=page%20and%20space=%27MUL%27%29&expand=version,history,history.lastUpdated"
    pagination:
       limit: "1000"
       start: "0"
```
And  then dynamically append these values during the GET on `api-call-confluence.xml` (recursive call subflow)

**Review by panizzag — COMMENTED** [review]
The changes look good. I couldn't test the SDLCs on a real scenario, but the rest is running fine in on-prem. Left a comment regarding confluence paging. We can discuss if we decide to move forward as-is and modify it later.

File: src/main/mule/sources/external/systems/apis/api-call-confluence.xml (hunk 2 above)
Comment (DavoCoder) [reply src/main/mule/sources/external/systems/apis/api-call-confluence.xml:35]: Correct, when testing, the pagination was causing an infinite loop (max recursive calls) - this change was just to break that loop. all the SDLC part requires improvement

**Review by panizzag — APPROVED** [review]

**Pull request merged**